src/core/variable.jl

```diff
@@ -522,6 +522,75 @@ function variable_storage_discharge(pm::GenericPowerModel; nw::Int=pm.cnw)
     )
 end
 
+"variables for modeling storage units, includes grid injection and internal variables, with mixed int variables for charge/discharge"
+function variable_storage_mi(pm::GenericPowerModel; kwargs...)
+    variable_active_storage(pm; kwargs...)
+    variable_reactive_storage(pm; kwargs...)
+    variable_storage_energy(pm; kwargs...)
+    variable_storage_charge(pm; kwargs...)
+    variable_storage_discharge(pm; kwargs...)
+    variable_storage_complementary_mi(pm; kwargs...)
+end
+
+""
+function variable_storage_complementary_mi(pm::GenericPowerModel; nw::Int=pm.cnw)
+    var(pm, nw)[:sc_on] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :storage)], base_name="$(nw)_sc", Bin,
+        start = comp_start_value(ref(pm, nw, :storage, i), "sc_on_start", 0)
+    )
+    var(pm, nw)[:sd_on] = JuMP.@variable(pm.model,
+        [i in ids(pm, nw, :storage)], base_name="$(nw)_sd", Bin,
+        start = comp_start_value(ref(pm, nw, :storage, i), "sd_on_start", 0)
+    )
+end
+
+function variable_storage_on_off(pm::GenericPowerModel; kwargs...)

```

Comment:
This should be `variable_storage_mi_on_off` to accommodate the NL variant where `variable_storage_complementary_indicator` is not required.

Reply:
@noahrhodes, not resolved?